### PR TITLE
begin diagnostics

### PR DESCRIPTION
**Pull request by ivanov**
so this has basic loading, but when I went to use the nipy timediff code, I hit a roadblock. There aren't wheels for python3.10, and even though I have numpy installed I couldn't build it

```
$ pip install nipy
Collecting nipy
  Using cached nipy-0.5.0.tar.gz (2.4 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [22 lines of output]
      Traceback (most recent call last):
        File "/home/pi/.pyenv/versions/xib/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/pi/.pyenv/versions/xib/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/pi/.pyenv/versions/xib/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-1xas1srr/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-1xas1srr/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-1xas1srr/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-1xas1srr/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 16, in <module>
        File "/tmp/pip-install-3dtndpos/nipy_d49ae1415ae541b399e8ec60d3aeea63/setup_helpers.py", line 35, in <module>
          from numpy.distutils.misc_util import appendpath
      ModuleNotFoundError: No module named 'numpy'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

which from https://github.com/nipy/nipy/issues/494 lead me to following https://github.com/KamitaniLab/bdpy/issues/68#issuecomment-1606582724 I was then able to resolve.

### CODE REVIEW

File: experiments/scripting_an_analysis.Rmd [hunk 1]
```diff
@@ -80,5 +80,18 @@ for subj_label, subject in study.subjects.items():
...
+plt.legend()
+```
 
+```{python}
+outliers.keys()
+```
 
+```{python}
+
+```
```
Comment (matthew-brett) [inline experiments/scripting_an_analysis.Rmd:97]:
```suggestion
```

File: xibabel/__init__.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import pathlib
 
 import nibabel as nib
+from nipy.algorithms.diagnostics.timediff import time_slice_diffs
```
Comment (matthew-brett) [inline xibabel/__init__.py:6]: What do you think - should this be part of xibabel's interface?

Comment (ivanov) [reply xibabel/__init__.py:6]: not sure what you mean

Comment (matthew-brett) [reply xibabel/__init__.py:6]: I mean - by importing in `__init__.py` you make it part of the public interface.

Comment (ivanov) [reply xibabel/__init__.py:6]: oh I see, then "no,  it should not be part of the public interface", I am just trying to get something going and figured we would probably even move most of this out of `__init__.py`,  just started sketching this out in what seemed like the most direct way.

Comment (ivanov) [reply xibabel/__init__.py:6]: but I guess if we were to leave it here for now, would you prefer we import it as `_time_slice_diffs` or should we explicitly make an `__all__` with things we want to be officially public, or hide the import inside the function that uses it?

Comment (matthew-brett) [reply xibabel/__init__.py:6]: Oh - it's fine - sorry - I was being anal.

**Pull request merged**